rocketpy/simulation/flight.py

```diff
@@ -1934,6 +1934,33 @@ def time(self):
         """Returns time array from solution."""
         return self.solution_array[:, 0]
 
+    def get_solution_at_time(self, t):
+        """Returns the solution state vector at a given time. If the time is
+        not found in the solution, the closest time is used and a warning is
+        raised.
+
+        Parameters
+        ----------
+        t : float
+            Time in seconds.
+
+        Returns
+        -------
+        solution : np.array
+            Solution state at time t. The array follows the format of the
+            solution array, with the first column being time like this:
+            [t, x, y, z, vx, vy, vz, e0, e1, e2, e3, omega1, omega2, omega3].
+
+        """
+        time_index = find_closest(self.time, t)
+        if abs(self.time[time_index] - t) > 1e-5:
```

Comment:
The way this is implemented here is good, but I would like to raise discussion about the following points:

- It might be convenient to have a `tol=1e-5` parameter;
- The counter argument would be that this parameter has no effect in code, only toggles a warning.

I mentioned this, because this value `1e-5` seems quite arbitrary. Is this warning really important?

Reply:
I just thought it would be wrong if the user sets a time but we take the state of a time close but not equal to the time he/she set.

I think adding this tolerance parameter could help, I will make it optional.

Reply:
Solved by [422bb51](https://github.com/RocketPy-Team/RocketPy/pull/441/commits/422bb51cf1ffcb1a586f79f7c063addaa103eb0b)